### PR TITLE
Update to released versions of patch dependencies.

### PR DESCRIPTION
**Pull request by nuttycom**
- bridgetree 0.3
- incrementalmerkletree 0.4
- orchard 0.5
- zcash_address 0.3
- zcash_note_encryption 0.4
- zcash_primitives 0.12
- zcash_proofs 0.12

Also, this updates the following dependencies to help eliminate duplicate transitive crate dependencies:
- ed25519-zebra 3.1
- metrics 0.21
- metrics-exporter-prometheus 0.12
- metrics-util 0.15

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -2413,7 +2395,7 @@ dependencies = [
 [[package]]
 name = "zcash_encoding"
 version = "0.2.0"
-source = "git+https://github.com/zcash/librustzcash.git?rev=36d7222685e65cd8b6335f67dfe619e1a8f13197#36d7222685e65cd8b6335f67dfe619e1a8f13197"
+source = "registry+https://github.com/rust-lang/crates.io-index"
```
Comment (daira) [inline Cargo.lock:2398]: I'm a bit hazy on how this is resolved. Does it use the version that was published as `zcash_encoding` 2.0 (which is earlier than 36d7222685e65cd8b6335f67dfe619e1a8f13197), or does it find a version of librustzcash that satisfies all the requested dependencies?

File: Cargo.lock [hunk 2]
```diff
@@ -2422,7 +2404,7 @@ dependencies = [
 [[package]]
 name = "zcash_history"
 version = "0.3.0"
-source = "git+https://github.com/zcash/librustzcash.git?rev=36d7222685e65cd8b6335f67dfe619e1a8f13197#36d7222685e65cd8b6335f67dfe619e1a8f13197"
+source = "registry+https://github.com/rust-lang/crates.io-index"
```
Comment (daira) [inline Cargo.lock:2407]: Same question as above.

**Review by daira — APPROVED** [review]
utACK with questions. Builds are failing because `Cargo.lock` needs to be updated, though (and some crates need to be vetted).

**Review by ebfull — APPROVED** [review]

**Review by daira — COMMENTED** [review]
Post-hoc ACK

**Pull request merged**